perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/SearcherBlImpl.java

```diff
@@ -84,6 +89,38 @@ public SearcherBlImpl(SearcherImplApi searcherImpl) {
 		return this.getUsersForCoreAttributesByMapOfAttributes(sess, mapOfCoreAttributesWithValues);
 	}
 
+	@Override
+	public List<Group> getGroups(PerunSession sess, Vo vo, Map<String, String> attributesWithSearchingValues) throws AttributeNotExistsException {
+		return this.getGroups(sess, attributesWithSearchingValues).stream()
+			.filter( group -> group.getVoId() == vo.getId())
+			.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<Group> getGroups(PerunSession sess, Map<String, String> attributesWithSearchingValues) throws AttributeNotExistsException {
+		//If there is no attribute, so every group match
+		if(attributesWithSearchingValues == null || attributesWithSearchingValues.isEmpty()) {
+			return perunBl.getVosManagerBl().getVos(sess).stream()
+				.flatMap( vo -> perunBl.getGroupsManagerBl().getAllGroups(sess, vo).stream())
+				.collect(Collectors.toList());
+		}
+
+		Map<Attribute, String> mapOfAttrsWithValues = new HashMap<>();
+		for(String name: attributesWithSearchingValues.keySet()) {
+			if(name == null || name.equals("")) throw new AttributeNotExistsException("There is attribute with no specific name!");
```

Comment:
I prefer isEmpty() rather than equals(""). It is more readable and I've read that isEmpty() is a little bit quicker than the other approach.

Reply:
Me too, copy&paste problem. Fixed.